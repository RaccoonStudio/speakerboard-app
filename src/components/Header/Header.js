import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { Link } from "gatsby"

const StyledHeader = styled.header``

const StyledSiteTitle = styled.h1``

const Header = ({ siteTitle }) => (
  <StyledHeader>
    <StyledSiteTitle>
      <Link to={`/app`}>{siteTitle}</Link>
    </StyledSiteTitle>
  </StyledHeader>
)

Header.propTypes = {
  /**
   * Specify the Header site title
   */
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
