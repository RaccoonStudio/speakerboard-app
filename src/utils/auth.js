import auth0 from "auth0-js"
import { navigate } from "gatsby"

const tokens = {
  accessToken: false,
  idToken: false,
  expiresAt: false,
}

const isBrowser = typeof window !== `undefined`

const getUser = () =>
  window.localStorage.gatsbyUser
    ? JSON.parse(window.localStorage.gatsbyUser)
    : {}

const setUser = user => (window.localStorage.gatsbyUser = JSON.stringify(user))

export const auth = isBrowser
  ? new auth0.WebAuth({
      domain: process.env.AUTH0_DOMAIN,
      clientID: process.env.AUTH0_CLIENTID,
      redirectUri: process.env.AUTH0_CALLBACK,
      responseType: "token id_token",
      scope: "openid profile email",
    })
  : {}

const setSession = () => (err, authResult) => {
  if (err) {
    navigate("/")
    return
  }

  if (authResult && authResult.accessToken && authResult.idToken) {
    let expiresAt = authResult.expiresIn * 1000 + new Date().getTime()
    tokens.accessToken = authResult.accessToken
    tokens.idToken = authResult.idToken
    tokens.expiresAt = expiresAt
    localStorage.setItem("isLoggedIn", true)
    navigate("/app")
    return setUser(authResult.idTokenPayload)
  }
}

export const handleAuthentication = () => {
  if (!isBrowser) {
    return
  }

  auth.parseHash(setSession())
}

export const isLoggedIn = () => {
  if (!isBrowser) return false

  return localStorage.getItem("isLoggedIn") === "true"
}

export const getCurrentUser = () => isBrowser && getUser()

export const logout = () => {
  if (!isBrowser) return

  navigate(`/logout`)
  setUser({})
  localStorage.setItem("isLoggedIn", false)
  auth.logout()
}

export const isAuthenticated = () => {
  if (!isBrowser) {
    return
  }

  return localStorage.getItem("isLoggedIn") === true
}

export const login = () => {
  if (!isBrowser) {
    return
  }

  auth.authorize()
}

export const silentAuth = callback => {
  if (!isAuthenticated()) return callback()

  auth.checkSession({}, setSession(callback))
}
