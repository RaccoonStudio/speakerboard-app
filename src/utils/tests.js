import renderer from "react-test-renderer"

export const render = ReactElement => {
  return renderer.create(ReactElement).toJSON()
}
