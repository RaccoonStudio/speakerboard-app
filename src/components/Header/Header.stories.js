import React from "react"
import { storiesOf } from "@storybook/react"
import { text } from "@storybook/addon-knobs"
import Header from "./Header"

const stories = storiesOf(`Header`, module)

stories.add(`overview`, () => (
  <Header siteTitle={text(`Site title`, undefined)}></Header>
))
