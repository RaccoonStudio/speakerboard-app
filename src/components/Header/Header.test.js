import React from "react"

import Header from "../Header"
import { render } from "../../utils/tests"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = render(<Header siteTitle="Test title" />)

    expect(tree).toMatchSnapshot()
  })
})
