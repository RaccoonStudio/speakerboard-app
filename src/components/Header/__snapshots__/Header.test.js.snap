// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className=""
>
  <h1
    className=""
  >
    <a
      href="/app"
    >
      Test title
    </a>
  </h1>
</header>
`;
